addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -49,6 +50,14 @@ def _get_specific_rendering_values(self, processing_values):
         converted_amount = payment_utils.to_minor_currency_units(self.amount, self.currency_id)
         base_url = self.provider_id.get_base_url()
         return_url_params = {'reference': self.reference}
+
+        # sanitize partner phone
+        country_code = self.partner_country_id.code
+        country_phone_code = self.partner_country_id.phone_code
+        phone = phone_sanitize_numbers([self.partner_phone], country_code, country_phone_code).get(self.partner_phone, {}).get('sanitized')
+        if not phone:
+            raise ValidationError("Razorpay: " + _("Incorrect phone number."))
```

Comment:
Also, `phone_sanitize_numbers` also send a `msg` of the error that could be helpful to know if the problem is an incorrect prefix, a missing number... I'd pass this error message instead of adding a generic one. The only case where you won't have a `msg` is when you don't have any numbers in the list, which, as said before, shouldn't be sanitized then.

Reply:
Following the test that show that without a phone number, there also is a problem, in this case, a `msg` could be crafted to explain that the partner doesn't have a phone number.